### PR TITLE
Move to .NET Core tooling

### PR DESCRIPTION
**Pull request by nayato**
Motivation
.NET Core tooling allows building for both .NET Framework and .NET Standard while easing pains of running tests and packaging.

Modifications
- csproj -> xproj + project.json
- support for .NET Standard 1.3
- build script is now based on CAKE
- using dotnet CLI for building, testing, packing
- strong name by default (snk included)

Result
DotNetty works on .NET Core and any other platform supporting .NET Standard 1.3+.

### CODE REVIEW

**Comment by nayato** [comment]
@mtuchkov PTAL

**Comment by nayato** [comment]
FYI @davidfowl @terrajobst

File: examples/Factorial/project.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  "frameworks": {
+    "netcoreapp1.0": {
+      "dependencies": {
+        "Microsoft.NETCore.App": {
+          "version": "1.0.0-*",
+          "type": "platform"
+        }
+      },
+      "imports": "dnxcore50"
+    }
```
Comment (nayato) [inline examples/Factorial/project.json]: I couldn't figure out how to make `net451` work here due to `BigInteger`. Adding dependency on `System.Numerics` won't work, with dependency on `System.Runtime.Numerics` still fails to build `net451`. cc @terrajobst,

**Comment by alexvaluyskiy** [comment]
@nayato maybe you should wait when MS will switch from project.json to MSBuild?

File: examples/Factorial/project.json (hunk 1 above)
Comment (alexvaluyskiy) [inline examples/Factorial/project.json]: Have you tried frameworkAssemblies?
```json
    "net451": {
      "frameworkAssemblies": {
        "System.Numerics": "4.0.0.0"
      }
    } 
```

**Comment by nayato** [comment]
@alexvaluyskiy that would be much easier change. Besides, they promise a migration tool :)
With .NET Core picking up now, doesn't make sense to wait more (another 2+ months?) until there is a stable tooling release.

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**